### DISCUSSION TITLE
AppleSilicon error - "Migration Failed"

### DISCUSSION THREAD

**Discussion by wildpika** [ppy/osu#17409] [Q&A] [answered]
![Screenshot 2022-03-22 at 15 01 23](https://user-images.githubusercontent.com/46168931/159499074-948063bd-8a18-4fa8-83f0-ca3a4ac8bee3.png)

Trying to start Osu on M1 Pro. Does anybody have any similar issues? 
This is the new version of Osu called osu.app.Apple.Silicon

**Comment by wildpika** [comment, marked as answer]
Temporary solution - download Intel version, extract, open Osu, wait for migration to be completed, exit and you can open Apple silicon version

**Comment by bdach** [comment]
please attach logs (https://github.com/ppy/osu/discussions/15188).

**Reply by wildpika** [reply]
they've been overwritten by successful Osu launch afterwards I suppose
[database.log](https://github.com/ppy/osu/files/8325782/database.log)
[network.log](https://github.com/ppy/osu/files/8325783/network.log)
[performance.log](https://github.com/ppy/osu/files/8325784/performance.log)
[runtime.log](https://github.com/ppy/osu/files/8325785/runtime.log)

**Reply by bdach** [reply]
they have been indeed. and as such they are not of much use anymore.

**Reply by wildpika** [reply]
As for the actual state I will just mark the temporary solution above as answer to this issue - it might help somebody else

**Comment by peppy** [comment]
It would be very helpful if you could reproduce the issue, by copying the backup of your database back into place and attempting migration again. Alternative, you can attach the database backup here and we can investigate further. You can find the backups in your osu! folder (there's a button to get there from the settings panel).

**Reply by wildpika** [reply]
Ok, I will try to reproduce that.

**Reply by Uniwhalpaca** [reply]
This issue has occurred to me as well. Here are the logs before I use the temporary fix wildpika suggested.
[database.log](https://github.com/ppy/osu/files/8356777/database.log)
[network.log](https://github.com/ppy/osu/files/8356778/network.log)
[performance.log](https://github.com/ppy/osu/files/8356779/performance.log)
[runtime.log](https://github.com/ppy/osu/files/8356780/runtime.log)

**Reply by peppy** [reply]
Thanks that clears things up.

The sqlite library we use is not providing an ARM dll so migration simply cannot work. Huge oversight. Might be best to remove m1 builds for now..

Did the migration immediately work after switching to an intel build and running it with no other changes?

**Reply by wildpika** [reply]
> Thanks that clears things up.
> 
> The sqlite library we use is not providing an ARM dll so migration simply cannot work. Huge oversight. Might be best to remove m1 builds for now..
> 
> Did the migration immediately work after switching to an intel build and running it with no other changes?

In my case it went instantly successfully after switching to Intel version and back to AppleSlcn after migration

**Reply by Uniwhalpaca** [reply]
It instantly worked for me as well. Please keep the m1 builds haha.

**Comment by inb4ohnoes** [comment]
I've just encountered this error as well. I've force quit and launched twice, same issue. Haven't tried the workaround mentioned above yet. Please see logs attached.

[database.log](https://github.com/ppy/osu/files/8366569/database.log)
[network.log](https://github.com/ppy/osu/files/8366570/network.log)
[performance.log](https://github.com/ppy/osu/files/8366571/performance.log)
[runtime.log](https://github.com/ppy/osu/files/8366572/runtime.log)

Edit: oops didn't see you already got them above. Never mind...

**Reply by peppy** [reply]
Please use the "workaround". We aren't fixing this any other way.

**Reply by inb4ohnoes** [reply]
Yep, workaround worked, thanks peppy
